### PR TITLE
[Travis] Test R on Travis

### PR DESCRIPTION
**Pull request by krlmlr**
- [Travis] Reorganization, now almost all instructions are part of the matrix
- [Travis] Need manual `./configure` to copy C++ files
- [Travis] Pseudo-env-var `TRAVIS_LANG` used to describe the builds in the Travis UI
- [Travis] Coverage checking, please visit https://codecov.io and sign up with GitHub
- Add README badges.
- [R] Run `document()`
- [R] Remove documentation of S3 methods
- [R] Improve test coverage, output and some corner cases still missing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,19 +27,28 @@ matrix:
...
+    - ctest
+    env: TRAVIS_LANG=C++
+  - os: linux
+    language: r
+    compiler: gcc
+    before_install:
+    - cd R
+    - ./configure
+    after_success:
+    - Rscript -e 'install.packages("covr"); covr::codecov()'
```
Comment (wesm) [inline .travis.yml]: does codecov have any notion of subprojects (so we could see the language-specific coverage?)?

**Comment by wesm** [comment]
Merged. thank you!

File: .travis.yml (hunk 1 above)
Comment (krlmlr) [inline .travis.yml]: Coverage is tracked separately per build job: [tibble](https://codecov.io/github/hadley/tibble) has three R versions in a matrix, offers unified and separate view.

Comment (wesm) [inline .travis.yml]: What I mean is, this is somewhat distracting for patches that don't affect the R bindings: https://github.com/wesm/feather/pull/66#issuecomment-204723657 . Are there any solutions?

Comment (krlmlr) [inline .travis.yml]: I think this becomes a non-issue if you add coverage analysis to the other languages. We may turn off R coverage for now to avoid this distraction, or perhaps you turn off the PR comments in codecov.

Comment (wesm) [inline .travis.yml]: I turned off the PR comments -- I'll add codecov.io for C++ and Python when I can as that will be nice to look at =)
